docs/docsite/rst/dev_guide/overview_architecture.rst

```diff
@@ -67,8 +75,70 @@ Here's what a simple playbook looks like::
     - common
     - content
 
-
-Extending Ansible with plug-ins and the API
+Extending Ansible with plugins and the API
 ===========================================
 
-Should you want to write your own, Ansible modules can be written in any language that can return JSON (Ruby, Python, bash, etc). Inventory can also plug in to any datasource by writing a program that speaks to that datasource and returns JSON. There's also various Python APIs for extending Ansible's connection types (SSH is not the only transport possible), callbacks (how Ansible logs, etc), and even for adding new server side behaviors.
+You can write your own specialized Ansible modules, module utilities, or plugins in any language that can return JSON (Ruby, Python, bash, etc). You can write an :ref:`inventory plugin <developing_inventory>` to connect to any datasource that returns JSON. There are also various Python APIs for extending Ansible's connection types (SSH is not the only transport possible), for adding callbacks (how Ansible logs, etc), and even for adding new server side behaviors.
+
+.. _ansible_search_path:
+
+The Ansible search path
+=======================
+
```

Comment:
Run this section by @nitzmahone to make sure it is correct and complete after his collections changes.

Reply:
This is all still accurate with collections, but collections adds a whole separate mechanism for role/plugin/etc loading that specifically does *not* grow or change throughout the run. I'm not sure how/where we want to document all that- I assume @acozine will come pick my brain when the time is right. The solution is probably a whole new "writing collections" page, as well as some mention/link in these pages.

Reply:
I'll add this to the 2.9 pile . . .